### PR TITLE
Removed tabs permission

### PR DESCRIPTION
**Pull request by OfficialNoob**
Removing use of tab.url as its not needed
Status: Awaiting Review

### CODE REVIEW

**Comment by ajayyy** [comment]
Can you use camelCase for function and variable names. (First letter lower case)

**Comment by OfficialNoob** [comment]
@ajayyy sure such rules can also be enforced with a linter

**Comment by OfficialNoob** [comment]
@ajayyy it should be done now it toke longer then expected because I tried onhashchange instead of tabUpdate did not work

**Comment by ajayyy** [comment]
`tabUpdate` did work?

**Comment by OfficialNoob** [comment]
Think so but i removed the function as all it needed to do was message the activeTab on chrome.tabs.onUpdated what I tried todo was detect the url change directly from the contentscript using onhashchanged witch did not work

**Comment by ajayyy** [comment]
I don't think that'll work since it has to be triggered by a user action

**Comment by OfficialNoob** [comment]
@ajayyy what will not work? this PR does not need tabs or activeTab permissions
by "activeTab " I meant the activeTab not the permission

**Comment by OfficialNoob** [comment]
"You can use most chrome.tabs methods and events without declaring any permissions in the extension's manifest file. However, if you require access to the url, title, or favIconUrl properties of tabs.Tab, you must declare the "tabs" permission in the manifest" https://developer.chrome.com/extensions/tabs

**Comment by OfficialNoob** [comment]
Did a test it seems to work fine please review

File: content.js [hunk 1]
```diff
@@ -1143,3 +1168,26 @@ function sendRequestToCustomServer(type, fullAddress, callback) {
   //submit this request
   xmlhttp.send();
 }
+
+function getYouTubeVideoID(url) { // Returns with video id else returns false
```
Comment (ajayyy) [inline content.js]: Don't add this back

File: popup.js [hunk 2]
```diff
@@ -201,16 +201,18 @@ function runThePopup() {
...
   
-  
   chrome.tabs.query({
       active: true,
       currentWindow: true
-  }, loadTabData);
+  }, onTabs);
   
+  function onTabs(tabs) {
+	 chrome.storage.sync.get(['videoid'], function(result) {
```
Comment (ajayyy) [inline popup.js]: change to `videoID`

File: background.js [hunk 3]
```diff
@@ -1,27 +1,10 @@
-//when a new tab is highlighted
-chrome.tabs.onActivated.addListener(
-  function(activeInfo) {
-    chrome.tabs.get(activeInfo.tabId, function(tab) {
-      let id = getYouTubeVideoID(tab.url);
+//the id of this user, randomly generated once per install
+var userID = null;
```
Comment (ajayyy) [inline background.js]: Don't add this back

File: background.js [hunk 4]
```diff
@@ -1,27 +1,10 @@
...
-      //if this even is a YouTube tab
-      if (id) {
-        videoIDChange(id, activeInfo.tabId);
-      }
-    })
-  }
-);
-
-//when a tab changes URLs
 chrome.tabs.onUpdated.addListener(function(tabId, changeInfo, tab) {
```
Comment (ajayyy) [inline background.js]: This can still be used without the permission?

File: content.js [hunk 5]
```diff
@@ -179,11 +179,36 @@ document.onkeydown = function(e){
...
   //not a url change
-  if (sponsorVideoID == id){
-    return;
-  }
+  if (sponsorVideoID == id) return;
+  
+  chrome.storage.sync.set({videoid: id});
+
+  //warn them if they had unsubmitted times
+  if (previousVideoID != null) {
```
Comment (ajayyy) [inline content.js]: Content script can do notifications?

File: background.js (hunk 4 above)
Comment (OfficialNoob) [reply background.js]: yes

File: content.js (hunk 1 above)
Comment (OfficialNoob) [reply content.js]: Removed.

File: popup.js (hunk 2 above)
Comment (OfficialNoob) [reply popup.js]: Done

File: content.js (hunk 5 above)
Comment (OfficialNoob) [reply content.js]: Not sure may need to create API I think the code works better per tab

Comment (OfficialNoob) [reply content.js]: Added alertPrevious message to background

**Comment by OfficialNoob** [comment]
Made requested code changes
Note: I think it would be good sometime to have videoID for each tab in the storage API

**Comment by ajayyy** [comment]
True, this will break if two tabs load at similar times. Can content scripts get their own tab id?

File: popup.js [hunk 6]
```diff
@@ -201,16 +201,18 @@ function runThePopup() {
...
   chrome.tabs.query({
       active: true,
       currentWindow: true
-  }, loadTabData);
+  }, onTabs);
   
+  function onTabs(tabs) {
+	 chrome.storage.sync.get(['videoID'], function(result) {
+         	loadTabData(tabs, result.videoid);
+	 });
```
Comment (ajayyy) [inline popup.js]: What is the point of this, why doesn't it send a message to the tab asking for the videoID?

File: popup.js [hunk 7]
```diff
@@ -201,16 +201,18 @@ function runThePopup() {
...
+  function onTabs(tabs) {
+	 chrome.storage.sync.get(['videoID'], function(result) {
+         	loadTabData(tabs, result.videoid);
+	 }); 
+  }
   
-  function loadTabData(tabs) {
-    //set current videoID
-    currentVideoID = getYouTubeVideoID(tabs[0].url);
+  function loadTabData(tabs, currentVideoID) {
```
Comment (ajayyy) [inline popup.js]: Is `tabs` still needed here?

Comment (OfficialNoob) [reply popup.js]: Currently Yes as tabs[0].id is used for isInfoFound

File: popup.js (hunk 6 above)
Comment (OfficialNoob) [reply popup.js]: There might be problems when switching between multiple tabs

- Make Storage API contain the tabID
- Send a message for the active tab (Probably best option)

**Comment by OfficialNoob** [comment]
Check now :/

File: popup.js (hunk 6 above)
Comment (OfficialNoob) [reply popup.js]: Went for "Send a message to active tab"

**Comment by ajayyy** [comment]
So, the tabs message from the popup still works without the tabs permission?

File: background.js [hunk 8]
```diff
@@ -1,27 +1,7 @@
-//when a new tab is highlighted
-chrome.tabs.onActivated.addListener(
```
Comment (ajayyy) [inline background.js]: Why was this removed?

Comment (ajayyy) [reply background.js]: This actually shouldn't be necessary anymore.
